pkg/operator/staticpod/controller/monitoring/monitoring_resource_controller.go

```diff
@@ -119,7 +120,13 @@ func (c MonitoringResourceController) sync() error {
 		errs = append(errs, fmt.Errorf("manifests/service-monitor.yaml: %v", err))
 	} else {
 		_, serviceMonitorErr := resourceapply.ApplyServiceMonitor(c.dynamicClient, c.eventRecorder, serviceMonitorBytes)
-		errs = append(errs, serviceMonitorErr)
+		// This is to handle 'the server could not find the requested resource' which occurs when the CRD is not available
+		// yet (the CRD is provided by prometheus operator). This produce noise and plenty of events.
+		if errors.IsNotFound(serviceMonitorErr) {
+			klog.V(4).Infof("Unable to apply service monitor: %v", err)
```

Comment:
when and how will this get re-kicked to create again?

Reply:
@deads2k 
```
// TODO: We need a dynamic informer here to observe changes to ServiceMonitor resource.
```

this should then be probably synthetic resync with no events reported until we get above.

Reply:
updated.